### PR TITLE
feats from zuul

### PR DESCRIPTION
**Pull request by AlessandroLorenzi**
- feat(web-be): deployment command support
- feat(be-web): cronjob volume and volume mount
- feat(be-web): hooks job volume and volume mount

### CODE REVIEW

File: charts/be-web/templates/cronjob.yaml [hunk 1]
```diff
@@ -84,6 +84,10 @@ spec:
               resources:
                 {{- toYaml . | nindent 16 }}
               {{- end }}
+              volumeMounts:
+              {{- with .volumeMounts }}
+                {{- toYaml . | nindent 16 }}
+              {{- end }}
```
Comment (giuscri) [inline charts/be-web/templates/cronjob.yaml:90]:
```suggestion
              {{- with .volumeMounts }}
              volumeMounts:
                {{- toYaml . | nindent 16 }}
              {{- end }}
```

File: charts/be-web/templates/cronjob.yaml [hunk 2]
```diff
@@ -97,5 +101,9 @@ spec:
           tolerations:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+          volumes:
+          {{- with .volumes }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
```
Comment (giuscri) [inline charts/be-web/templates/cronjob.yaml:107]:
```suggestion
          {{- with .volumes }}
          volumes:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

File: charts/be-web/templates/hooks/job.yaml [hunk 3]
```diff
@@ -76,6 +76,10 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+          volumeMounts:
+          {{- with .volumeMounts }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
```
Comment (giuscri) [inline charts/be-web/templates/hooks/job.yaml:82]:
```suggestion
          {{- with .volumeMounts }}
          volumeMounts:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

File: charts/be-web/templates/hooks/job.yaml [hunk 4]
```diff
@@ -89,4 +93,8 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      volumes:
+      {{- with .volumes }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (giuscri) [inline charts/be-web/templates/hooks/job.yaml:99]:
```suggestion
      {{- with .volumes }}
      volumes:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

**Review by giuscri — CHANGES_REQUESTED** [review]

**Review by gi0baro — CHANGES_REQUESTED** [review]
@AlessandroLorenzi can you please provide use cases (even with links from zuul repo) for:
- deployment command: why you can't use the entrypoint like everything else?
- mounting volumes: where does these volumes come from? mounting such things in hooks is quite dangerous as it might be executed in transient states where those data doesn't exist (yet)

Also, in general, given the changes, I would make a minor, not a patch version

**Comment by AlessandroLorenzi** [comment]
> @AlessandroLorenzi can you please provide use cases (even with links from zuul repo) for:
> 
>     * deployment command: why you can't use the entrypoint like everything else?

You are right, I could recreate the container. Anyway, command is supported by k8s, so why not?

> 
>     * mounting volumes: where does these volumes come from? mounting such things in hooks is quite dangerous as it might be executed in transient states where those data doesn't exist (yet)
> 

in zuul:

```yaml
      volumes:
      - name: gdrive-service-account
        secret:
          secretName: gdrive-service-account
          optional: false
```

I'ts managed by `kubernetes-app-secrets` repo, so no problem with that.


> 
> Also, in general, given the changes, I would make a minor, not a patch version

👍🏼

**Pull request closed without merging**